.github/workflows/copilot_deploy.yml

```diff
@@ -43,6 +43,25 @@ jobs:
       check_db_migrations: true
       assets_required: true
 
+  type_checks:
+    needs: [setup]
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
+      - name: Set up Python
+        uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5
+        with:
+          python-version: 3.10.x
+      - name: Install uv
+        uses: astral-sh/setup-uv@887a942a15af3a7626099df99e897a18d9e5ab3a # v5
+        with:
+          enable-cache: true
+      - name: install dependencies
+        run: uv sync
+      - name: run type checking
+        run: uv run mypy
```

Comment:
:chefkiss: nice - need to do this on post-award too because otherwise you have to source the venv.